fblackburn1/node-red-contrib-hubitat
# Global cache

Ready to be merged

### manual tests

Scenarios that should be tested manually before merging this branch

1. deploy device node, new device HE, deploy config node
    * OK: Cache is built on deploy
2. deploy device node, old device HE, without redeploy config node
    * OK: deploying device node will redeploy config node automatically
3. deploy device node, old device HE, deploy config node
    * OK: Same as 1
4. deploy device node, new device HE, without redeploy config node
    * OK: It will be initialized like other nodes
5. redeploy other nodes, without redeploy config and device node
    * OK
6. deploy config node, without redeploy device node
    * OK: if config node is redeployed, then all related nodes will be redeployed too

7. missing HE events
    * OK: on the next event received, the state will be restored

### Note
All device nodes related to a configuration node will be redeployed even if you only deploy `Modified Nodes` when you will update one of these node (ex: if you have config node A and device node B & C, then if you update only node C and deploy modified flow, then node A-B-C will be deployed). That's because the config node related to the modified node will be mark as dirty and will dirt all related nodes.
This behavior seems only related to device node (not other nodes)
